src/test/QuantumGravityBridgeBenchmark.t.sol

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.19;
+
+import "openzeppelin-contracts/contracts/utils/cryptography/ECDSA.sol";
+
+import "../Constants.sol";
+import "../DataRootTuple.sol";
+import "../QuantumGravityBridge.sol";
+import "../lib/tree/binary/BinaryMerkleProof.sol";
+
+import "ds-test/test.sol";
+
+interface CheatCodes {
+    function addr(uint256 privateKey) external returns (address);
+    function sign(uint256 privateKey, bytes32 digest) external returns (uint8 v, bytes32 r, bytes32 s);
+    function deriveKey(string calldata, string calldata, uint32) external returns (uint256);
+}
+
+/// @notice Example command to run the benchmark:
+/// `forge test --match-test testBenchmarkSubmitDataRootTupleRoot -vvvvvv --gas-report`.
+/// To change the validator set size, change the `numberOfValidators` constant.
+/// To make custom calculations of the gas, you can use the `gasleft()` solidity
+/// built-in function.
+/// The following answer has some insights on using that:
+/// https://ethereum.stackexchange.com/a/132325/65649
+/// The gas estimations might not be accurate to the real cost in a real network,
+/// and that's because foundry doesn't track calldata cost. source:
+/// https://github.com/foundry-rs/foundry/issues/3475#issuecomment-1469940917
+/// To have accurate results, make sure to add the following costs:
+/// A byte of calldata costs either 4 gas (if it is zero) or 16 gas (if it is any other value).
+contract Benchmark is DSTest {
```

Comment:
[note for reviewers]
This is always run, but it only takes `5.71ms`, so I guess it's okey. we can leave it running with the tests.